pkg/manager/rollbackconfig.go

```diff
@@ -105,6 +110,58 @@ func computeRollback(m *Manager, target string, configname store.ConfigName) (ch
 	return id, updates, deletes, nil
 }
 
+//Deprecated computeRollback works on old non atomix methods
+func computeNewRollback(m *Manager, rollbackChange *networkchangetypes.NetworkChange) (*networkchangetypes.NetworkChange, error) {
+	deltaChange := &networkchangetypes.NetworkChange{
+		ID: networkchangetypes.ID(namesgenerator.GetRandomName(0)),
+		Status: changetypes.Status{
+			Phase:   changetypes.Phase_ROLLBACK,
+			State:   changetypes.State_PENDING,
+			Reason:  changetypes.Reason_NONE,
+			Message: "Change generated for administratively requested rollback of " + string(rollbackChange.ID),
+		},
+	}
+	prevChanges := make([]*devicechangetypes.Change, 0)
+	for _, deviceChange := range rollbackChange.Changes {
+		previousValues := make([]*devicechangetypes.ChangeValue, 0)
+		for _, value := range deviceChange.Values {
+			preVal, err := m.GetTargetNewConfig(string(deviceChange.DeviceID), value.Path, 1)
```

Comment:
We might want to consider doing reverse iteration to get the previous value for a path instead of reading up to the previous change for the path. We should expect the change lists to get pretty large. We can even add a reverse iteration method to stores to stream previous changes from Raft nodes. I don’t think it’s something that needs to be fixed here, but we probably want to make a todo to optimize the way this computation is done since it’s coming from distributed stores.

Reply:
introducing a TODO